### ISSUE TITLE
linter: `typescript/prefer-for-of` misses nested computed member expressions

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26565]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
    "rules": {
        "typescript/prefer-for-of": "error"
    }
}
```

### What happened?

```ts
class Example {
    private readonly values: Record<string, number[]> = {};

    print(key: string) {
        for (let index = 0; index < this.values[key].length; index++) {
            console.log(this.values[key][index]);
        }
    }
}
```

## Actual result

Oxlint reports no violation.

## Expected result

Oxlint should report the loop because `index` is used only to read elements from the same
`this.values[key]` collection:

```ts
for (const value of this.values[key]) {
    console.log(value);
}
```

`@typescript-eslint/prefer-for-of` 8.69.0 reports this pattern.
